population/client_test.go

```diff
@@ -530,6 +530,171 @@ func TestGetPopulationTypes(t *testing.T) {
 	})
 }
 
+func TestGetAreaTypesParent(t *testing.T) {
+	const userAuthToken = "user"
+	const serviceAuthToken = "service"
+	const datasetId = "datasetId"
+	const areaTypeId = "areaId"
+	Convey("Given a valid request", t, func() {
+		stubClient := newStubClient(&http.Response{Body: ioutil.NopCloser(bytes.NewReader(nil))}, nil)
+		client, err := NewWithHealthClient(health.NewClientWithClienter("", "http://test.test:2000/v1", stubClient))
+		So(err, ShouldBeNil)
+
+		input := GetAreaTypesParentsInput{
+			UserAuthToken:    userAuthToken,
+			ServiceAuthToken: serviceAuthToken,
+			DatasetID:        datasetId,
+			AreaTypeID:       areaTypeId,
+		}
+		_, _ = client.GetAreaTypesParents(context.Background(), input)
+
+		Convey("it should call the area types parens endpoint", func() {
+			calls := stubClient.DoCalls()
+			So(calls, ShouldNotBeEmpty)
+			So(calls[0].Req.URL.String(), ShouldEqual, "http://test.test:2000/v1/population-types/datasetId/area-types/areaId/parents")
+		})
+	})
+
+	Convey("Given authentication tokens", t, func() {
+		stubClient := newStubClient(&http.Response{Body: ioutil.NopCloser(bytes.NewReader(nil))}, nil)
+		client := newHealthClient(stubClient)
+
+		input := GetAreaTypesParentsInput{
+			UserAuthToken:    userAuthToken,
+			ServiceAuthToken: serviceAuthToken,
+		}
+
+		_, _ = client.GetAreaTypesParents(context.Background(), input)
+
+		Convey("it should set the auth headers on the request", func() {
+			calls := stubClient.DoCalls()
+			So(calls, ShouldNotBeEmpty)
+
+			So(calls[0].Req, shouldHaveAuthHeaders, userAuthToken, serviceAuthToken)
+		})
+	})
+
+	Convey("Given a valid areaTypes parents response payload", t, func() {
+		parents := GetAreaTypesParentsResponse{
+			AreaTypes: []AreaTypes{{ID: "test", Label: "Test", TotalCount: "2"}},
+		}
+
+		resp, err := json.Marshal(parents)
+		So(err, ShouldBeNil)
+
+		stubClient := newStubClient(&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader(resp)),
+		}, nil)
+		client := newHealthClient(stubClient)
+
+		input := GetAreaTypesParentsInput{
+			UserAuthToken:    userAuthToken,
+			ServiceAuthToken: serviceAuthToken,
+		}
+		res, err := client.GetAreaTypesParents(context.Background(), input)
+
+		Convey("it should return a list of population types", func() {
+			So(err, ShouldBeNil)
+			So(res, ShouldResemble, parents)
+		})
+	})
+
+	Convey("Given the area types parents API returns an error", t, func() {
+		stubClient := newStubClient(nil, errors.New("oh no"))
+
+		client := newHealthClient(stubClient)
+
+		input := GetAreaTypesParentsInput{
+			UserAuthToken:    userAuthToken,
+			ServiceAuthToken: serviceAuthToken,
+		}
+		_, err := client.GetAreaTypesParents(context.Background(), input)
+
+		Convey("it should return an internal error", func() {
+			So(err, shouldBeDPError, http.StatusInternalServerError)
+		})
+	})
+	/* Currently the swagger spec in dp-population-type-api doesn't expect any error codes
```

Comment:
You should probably uncomment these tests or remove them. The only hard-coded status code in the endpoint is 500 but any status code returned from Cantabular can also be returned. And there are many reasons a status code could be returned other than what is defined in the swagger spec, if Cantabular gets updated or the service is unreachable etc